src/flamenco/vm/fd_vm_interp_core.c

```diff
@@ -163,7 +163,7 @@
   shadow[ frame_cnt ].r9 = reg[9];                                                                          \
   shadow[ frame_cnt ].pc = pc;                                                                              \
   if( FD_UNLIKELY( ++frame_cnt>=frame_max ) ) goto sigstack; /* Note: untaken branches don't consume BTB */ \
-  reg[10] += FD_VM_STACK_FRAME_SZ + FD_VM_STACK_GUARD_SZ
```

Comment:
We ideally shouldn't call slot_ctx from the actual interpreter. the better approach is probably to add a field into the vm indicating if direct mapping is enabled or not

Reply:
Also we shouldn't probably add a conditional here if it can be avoided. Let's determine the stack guard size when the vm is instantiated